app/org/sagebionetworks/bridge/services/email/MimeTypeEmailProvider.java

```diff
@@ -1,9 +1,39 @@
 package org.sagebionetworks.bridge.services.email;
 
+import java.util.Set;
+
 import javax.mail.MessagingException;
 
-public interface MimeTypeEmailProvider {
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.models.studies.Study;
+
+import com.google.common.collect.Iterables;
+
+public abstract class MimeTypeEmailProvider {
 
-    MimeTypeEmail getMimeTypeEmail() throws MessagingException;
+    private Study study;
+    
+    MimeTypeEmailProvider(Study study) {
+        this.study = study;
+    }
```

Comment:
Minor: Should this be protected? If for whatever reason, the subclasses are in a different package.